packages/components/src/__future__/Tag/RemovableTag/subcomponents/RemoveButton.module.scss

```diff
@@ -0,0 +1,78 @@
+@import "~@kaizen/design-tokens/sass/border";
+@import "~@kaizen/design-tokens/sass/color";
+@import "../../../../../styles/button-reset";
+
+.removeButton {
+  @include button-reset;
+
+  position: relative;
+  display: inline-flex;
+  border-radius: 50%;
+
+  &:focus {
+    outline: none;
+  }
+
+  &:focus-visible {
+    &::after {
+      $focus-ring-offset: -3px;
+
+      border-color: $color-blue-500;
+      content: "";
+      position: absolute;
+      background: transparent;
+      border-width: $border-focus-ring-border-width;
+      border-style: $border-focus-ring-border-style;
+      border-radius: 50%;
+      inset: $focus-ring-offset;
+    }
+  }
+
+  &::before {
+    position: absolute;
+    content: "";
+    inset: -5px;
+    border-radius: 50%;
+  }
+
+  .closeIcon {
+    fill: rgba($color-purple-800-rgb, 0.7);
+  }
+
+  &:hover,
+  &:focus-visible {
+    cursor: pointer;
+
+    .closeIcon {
+      fill: $color-purple-800;
+    }
+
+    &.gray {
+      background-color: rgba($color-gray-600-rgb, 0.2);
+    }
+
+    &.blue {
+      background-color: rgba($color-blue-500-rgb, 0.3);
+    }
+
+    &.green {
+      background-color: rgba($color-green-500-rgb, 0.3);
+    }
+
+    &.yellow {
+      background-color: rgba($color-yellow-500-rgb, 0.5);
+    }
+
+    &.orange {
+      background-color: rgba($color-orange-500-rgb, 0.3);
+    }
+
+    &.red {
+      background-color: rgba($color-red-500-rgb, 0.3);
+    }
+
+    &.purple {
+      background-color: rgba($color-purple-500-rgb, 0.3);
+    }
+  }
```

Comment:
I think it's important we start to avoid nesting in general as the CSS generated will be more dense and we need to start being mindful of how large our CSS bundle is now that we compile it into a single file.

The output comparison:

```
// Nested: 240ch
.RemoveButton-module_removeButton__9yzxJ:focus-visible.RemoveButton-module_gray__M3vzs,
.RemoveButton-module_removeButton__9yzxJ:hover.RemoveButton-module_gray__M3vzs {
  background-color: rgba(var(--color-gray-600-rgb, 82, 78, 86), 0.2);
}

// Unnested: 153ch
.RemoveButton-module_gray__M3vzs:focus-visible,
.RemoveButton-module_gray__M3vzs:hover {
  background-color:rgba(var(--color-gray-600-rgb,82,78,86),.2)
}
```

And if we were to reduce all our component's CSS by this much, we'd get better overall performance.

Reply:
Cool, thanks for explaining. They're pulled out.